memif-attach/main.go

```diff
@@ -18,61 +18,112 @@ package main
 
 import (
 	"context"
+	"net"
 	"time"
-
+	// "io"
+	"encoding/binary"
+	"bufio"
+	"bytes"
 	"git.fd.io/govpp.git/api"
 	"github.com/edwarnicke/govpp/binapi/memif"
+	interfaces "github.com/edwarnicke/govpp/binapi/interface"
+	"github.com/edwarnicke/govpp/binapi/interface_types"
 	"github.com/edwarnicke/log"
 	"github.com/edwarnicke/vpphelper"
 )
 
+type MemifMsg struct {
+	HelloMsgType HelloMsg
+	InitMsgType InitMsg
+}
+// HelloReplyMsg type
+type HelloMsg struct {
+	Name [32]byte // 32 bytes array
+	Min_version uint16 
+	Max_version uint16
+	Max_region uint16 
+	Max_m2s_ring uint16 
+	Max_s2m_ring uint16
+	Max_log2_ring_size uint8 
+} 
+
+// InitMsg
+const MEMIF_SECRET_SIZE = 24
+type MemifInterfaceMode int
+const (
+	MEMIF_INTERFACE_MODE_ETHERNET MemifInterfaceMode = iota
+	MEMIF_INTERFACE_MODE_IP
+	MEMIF_INTERFACE_MODE_PUNT_INJECT
+)
+type InitMsg struct {
+	Version uint16 // check the file
+	Id uint32 // socket id?
+	Mode MemifInterfaceMode
+	Secret  [MEMIF_SECRET_SIZE]uint8
+	Name [32]uint8
```

Comment:
You may want to change the type of Name to [32]byte to match the Name field type from Hello.